src/CacheInterface.php

```diff
@@ -2,12 +2,22 @@
 
 namespace React\Cache;
 
+use React\Promise\PromiseInterface;
+
 interface CacheInterface
 {
-    // @return React\Promise\PromiseInterface
+    /**
+     * @return PromiseInterface
+     */
     public function get($key);
 
+    /**
+     * @return PromiseInterface
+     */
     public function set($key, $value);
 
+    /**
+     * @return PromiseInterface
```

Comment:
I'm not sure if `true` (or `bool` more generally) is a good idea. In [Redis cache](https://github.com/WyriHaximus/reactphp-cache-redis), the returning values are responses directly from Redis server (int, string, etc.). In example, for successful save operation "OK" string is returned as Redis returns it.

This is a general problem with `Promises`, that function should be described with `PromiseInterface` (to make IDE life easier and ours too) but we don't have any knowledge then about the value returned by this `Promise` itself when fulfilled. I saw in few libraries that declared return value was: `PromiseInterface|int|string`. We can simplify this to `PromiseInterface|mixed`. This will help code completition ans also give some tip about value retrieved from `Promise` itself. Implementing libraries can then narrow the return value to `PromiseInterface|string` or `PromiseInterface|MyGreatClass`.

Just a suggestion.